src/DynamoCore/Library/TypedParameter.cs

```diff
@@ -5,6 +5,19 @@
 
 namespace Dynamo.Library
 {
+    public struct TypedParameterParams
+    {
+        public TypedParameterParams(string parameterName, ProtoCore.Type type) : 
+            this()
+        {
+            ParameterName = parameterName;
+            Type = type;
+        }
+
+        public string ParameterName { get; set; }
+        public ProtoCore.Type Type { get; set; }
+        public object DefaultValue { get; set; }
+    }
```

Comment:
I thought so too, just hated methods with large number of arguments (I was about to add another one in a separate branch, and the list grows). Do you think this is not needed (or too small to consider harmful)? :)